pkg/pushsync/pushsync.go

```diff
@@ -54,8 +57,8 @@ func New(o Options) *PushSync {
 		quit:          make(chan struct{}),
 	}
 
-	go ps.chunksWorker()
-
+	ctx := context.Background()
+	go ps.chunksWorker(ctx)
```

Comment:
This context is deliberately moved in the previous pr to the chunkWorker function. Why is it here again?

Reply:
@janos Thanks for the input. My IDE was not configured for auto formatting and that was causing the issue. I fixed it now and even if it manually forget to format it before submission, autoformat should do the trick.

Reply:
We need contexts downstream anyway even if we dont use them.  I moved it to chunksworker according to your comment.

Reply:
Thanks, yes, this is how it is on master. Context is created in chunkWorker.